### DISCUSSION TITLE
Is Nginx used and working ?

### DISCUSSION THREAD

**Discussion by Etisti** [fastapi/full-stack-fastapi-template#1814] [Questions] [answered]
Hi, i would like to know, is Nginx used ?

- By front end ? => full-stack-fastapi-postgresql/{{cookiecutter.project_slug}}/frontend/nginx-backend-not-found.conf for wich purpose ?

- Is it used as reverse proxy and/or load balancer too ? 

- If used why using Nginx AND Traefik ???

Thanks a lot for thoses informations

**Comment by karolzlot** [comment]
Only Traefik is used as reverse proxy.

Nginx is used inside frontend docker container. I am sure nginx is not used anywhere else.

**Comment by Etisti** [comment]
why it is used inside frontend docker container and not has reverse proxy ? 
I don't understand the purpose of having 2 reverse proxy here ^^'

**Comment by karolzlot** [comment, marked as answer]
Nginx is not only reverse proxy, it is also a web server.

In this case it is configured to be used as a webserver to serve react app, you can see it in [nginx.conf](https://github.com/tiangolo/node-frontend/blob/master/nginx.conf).

The only reverse proxy used in this project is traefik. 

I think traefik is more modern than ngnix, so it was chosen for this project.

**Comment by Etisti** [comment]
I don't understand really well... So how many servers are set up on this project ?
Does the front end has his own server so it is like an other app communicating with the fastAPI app ? And traefik load balances the (how many?) gunicorns/uvicorns serveurs ?
I'm having a bad time trying to vizualise this I'm sorry 😅

**Comment by karolzlot** [comment]
Each container is a server. You can run reach container separately. 

I think you need to read more about docker/ docker compose and practice on simpler projects first

**Comment by Etisti** [comment]
I know about docker and compose.
It is just that I don't understand that you separate one serveur to serve for front end and an other one for back end... It is the architecture that I don't understand well

**Comment by shoaib30** [comment]
It actually is quite a standard architecture. It allows scaling the backend independently of the front end. And the load balancing can be for the backend services (which could be more than one)
